r3f/src/ZestyBanner.js

```diff
@@ -1,58 +1,75 @@
-import * as THREE from 'three'
-import { useLoader, useThree } from "@react-three/fiber"
-import { useRef, useState, Suspense, useEffect } from "react"
-import { fetchNFT, fetchActiveBanner, sendMetric } from "../../utils/networking"
-import { formats, defaultFormat } from '../../utils/formats';
-import { Interactive } from '@react-three/xr'
-import { parseProtocol } from '../../utils/helpers';
+import * as THREE from "three";
+import { useLoader, useThree } from "@react-three/fiber";
+import { useRef, useState, Suspense, useEffect } from "react";
+import {
+  fetchNFT,
+  fetchActiveBanner,
+  sendMetric,
+} from "../../utils/networking";
+import { formats, defaultFormat, defaultStyle } from "../../utils/formats";
+import { Interactive } from "@react-three/xr";
+import { parseProtocol } from "../../utils/helpers";
+
+export * from "../../utils/formats";
 
 export default function ZestyBanner(props) {
-  const [bannerData, setBannerData] = useState(false)
+  const [bannerData, setBannerData] = useState(false);
 
   const space = props.space ? props.space : props.adSpace;
-  const format = props.format ? props.format : props.adFormat;
-  console.log(format);
+  const format =
+    (props.format ? props.format : props.adFormat) ?? defaultFormat;
+
+  const width = props.width ?? formats[format].width;
+  const height = props.height ?? formats[format].height;
+
+  const newStyle = props.style ?? defaultStyle;
 
   const loadBanner = async (space, creator, network, format, style) => {
     const activeNFT = await fetchNFT(space, creator, network);
     const activeBanner = await fetchActiveBanner(activeNFT.uri, format, style);
     return activeBanner;
-  }
+  };
 
-  useEffect(() => {    
-    loadBanner(space, props.creator, props.network, format, props.style).then((data) => {
-      let banner = data.data;
-      let url = banner.url || banner.properties?.url;
-      if (url == 'https://www.zesty.market') {
-        url = `https://app.zesty.market/space/${props.space}`;
+  useEffect(() => {
+    loadBanner(space, props.creator, props.network, format, newStyle).then(
+      (data) => {
+        let banner = data.data;
+        let url = banner.url || banner.properties?.url;
+        if (url == "https://www.zesty.market") {
+          url = `https://app.zesty.market/space/${props.space}`;
+        }
+        banner.image = banner.image.match(/^.+\.(png|jpe?g)/i)
+          ? banner.image
+          : parseProtocol(banner.image);
+        sendMetric(
```

Comment:
I think we can comment this out for now, since we don't have anywhere for the metrics to go just yet.